tests/vm_watch_test.go

```diff
@@ -0,0 +1,334 @@
+		}
+	}
+
+	// Reads from stdin until a newline character is found
+	readLine := func(rc io.ReadCloser, timeout time.Duration) string {
+		lineChan := make(chan string)
+
+		go func() {
+			defer GinkgoRecover()
+
+			var line strings.Builder
+			buf := make([]byte, 1)
+			defer close(lineChan)
+
+			for {
+				n, err := rc.Read(buf)
+
+				if err != nil && err != io.EOF && !strings.Contains(err.Error(), "file already closed") {
+					Fail(err.Error())
+					break
+				}
+
+				if n > 0 {
+					if buf[0] != '\n' {
+						line.WriteByte(buf[0])
+					} else {
+						break
+					}
+				}
+			}
+
+			lineChan <- line.String()
+		}()
+
+		select {
+		case line := <-lineChan:
+			return line
+		case <-time.After(timeout):
+			err := rc.Close()
+			Expect(err).To(BeNil())
+
+			Fail("Timeout reached on read operation")
+
+			return ""
+		}
+	}
+
+	// Reads VM status from the given pipe (stdin in this case) and
+	// returns a new status.
+	// if old_status is non-nil, the function will read status lines until
+	// new_status.running != old_status.running in order to skip duplicated status lines
+	readVMStatus := func(rc io.ReadCloser, old_status *vmStatus, timeout time.Duration) *vmStatus {
+		new_stat := newVMStatus(strings.Fields(readLine(rc, timeout)))
+
+		for old_status != nil && new_stat.running == old_status.running {
+			new_stat = newVMStatus(strings.Fields(readLine(rc, timeout)))
+		}
+
+		return new_stat
+	}
+
+	// Reads VMI status from the given pipe (stdin in this case) and
+	// returns a new status.
+	// if old_status is non-nil, the function will read status lines until
+	// new_status.phase != old_status.phase in order to skip duplicated lines
+	readVMIStatus := func(rc io.ReadCloser, old_status *vmiStatus, timeout time.Duration) *vmiStatus {
+		newStat := newVMIStatus(strings.Fields(readLine(rc, timeout)))
+
+		for old_status != nil && newStat.phase == old_status.phase {
+			newStat = newVMIStatus(strings.Fields(readLine(rc, timeout)))
+		}
+
+		return newStat
+	}
+
+	// Create a command with output/error redirection.
+	// Returns (cmd, stdout, stderr)
+	createCommandWithNSAndRedirect := func(namespace, cmdName string, args ...string) (*exec.Cmd, io.ReadCloser, io.ReadCloser) {
+		cmdName, cmd, err := CreateCommandWithNS(namespace, cmdName, args...)
+
+		Expect(cmdName).To(Not(Equal("")))
+		Expect(cmd).To(Not(BeNil()))
+		Expect(err).To(BeNil())
+
+		// Output redirection
+		stdOut, err := cmd.StdoutPipe()
+		Expect(err).To(BeNil())
+		Expect(stdOut).To(Not(BeNil()))
+
+		stdErr, err := cmd.StderrPipe()
+		Expect(err).To(BeNil())
+		Expect(stdErr).To(Not(BeNil()))
+
+		return cmd, stdOut, stdErr
+	}
+
+	var virtCli kubecli.KubevirtClient
+	var err error
+
+	if kubeconfig := os.Getenv("KUBECONFIG"); len(kubeconfig) > 0 {
+		virtCli, err = kubecli.GetKubevirtClientFromFlags("", kubeconfig)
+	}
+
+	tests.PanicOnError(err)
+
+	// Define a timeout for read opeartions in order to prevent test hanging
+	readTimeout := 20 * time.Second
```

Comment:
Maybe those should be constants at the top of the test?

Reply:
I agree, fixed.